lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -480,6 +485,173 @@ def post(self, request):
+
+                    # getting other student enrollments
+                    enrollments = list(get_course_enrollments(u, course_org_filter, org_filter_out_set))
+                    enrolled_ids = ', '.join([enrollment.course_id.course for enrollment in enrollments])
+
+                    # Comparing course structure chunks (problem, video, html) with units visited in course
+                    # Need to get positions for sequentials because this is the
+                    # only one method to check whether student visit html-only lesson or no
+                    seq_positions = {}  # getting positions for sequential module_types
+                    visited_in_seq = db_query.filter(module_type='sequential')
+                    for v in visited_in_seq.values('module_state_key', 'state'):
+                        # structure example: v = {
+                        # 'module_state_key': u'block-v1:edX+DemoX+Demo_Course+type@sequential+block@edx_introduction',
+                        # 'state': u'{"position": 1}'
+                        # }
+                        key = v['module_state_key']
+                        val = json.loads(v['state']).get('position')
+                        seq_positions[key] = val
+
+                    status_list = []
+                    # Need to go through parent-children structure again for catching html-only lessons
+                    for sequential in sequentials:
+                        for vertical in get_children(sequential):
+                            for unit in get_children(vertical):
+                                if course_ordered[unit]['block_type'] in ['video', 'problem']:
+                                    if unit in visited_in_course:
+                                        status = '+'
+                                        break   # if we have visit in any of vertical children - it marks '+'
+                                # skipping discussion units
+                                elif course_ordered[unit]['block_type'] in ['discussion']:
+                                    continue
+                                # for html-only pages:
+                                else:
+                                    current_seq_children_list = get_children(sequential)
+                                    vert_position_in_seq = current_seq_children_list.index(vertical)
+                                    try:
+                                        last_visit_in_seq_position = seq_positions[sequential]
+                                        if (vert_position_in_seq < last_visit_in_seq_position):
+                                            status = '+'
+                                            break
+                                        else:
+                                            status = '-'
+                                    except Exception:
+                                        status = '-'
+                            status_list.append(status)
```

Comment:
due to so many indentations it is very hard to understand when this line gets executed. If you must have 3 loops, you should at least extract the code inside the loop into separate function. This way this code would be easier to understand.